### DISCUSSION TITLE
Get overlay port location from within an overlay triplet

### DISCUSSION THREAD

**Discussion by netcorefan1** [microsoft/vcpkg#31077] [Q&A] [answered]
I have been forced to open this discussion because after hours of trials I didn't found a way to achieve what should be a very simple task.
In an overlay triplet, I can get the current overlay port name from `${PORT}` variable. I also need its location, just like displayed in the log:
```
-- [OVERLAY] Loading triplet configuration from: C:\custom-triplets\custom.cmake
-- Installing port from location: C:\custom-port\xxx
-- Detected current port name: xxx
-- Detected current port location: ???
```

Thanks

**Comment by autoantwort** [comment]
The location is not available.

**Reply by netcorefan1** [reply]
I have just discovered (by pure luck) an undocumented (and well hidden from google searches) CURRENT_PORT_DIR variable. It correctly shows the port location, whether it is an overlay or not. Unless there is some warning on its usage that one should be aware of, I'll close the discussion as resolved.

**Reply by autoantwort** [reply]
But afaik this variable is not available inside a triplet while doing abi hashing/getting variables.

**Reply by dg0yt** [reply, marked as answer]
Here is the big fat

# WARNING

vcpkg tries to carefully to track the inputs which go into a binary artifact for a given port. This is known as ABI hashing here. The inputs include:
- The file content of the triplet file.
- The content of the files in the port directory.
- more ... cf. https://learn.microsoft.com/en-us/vcpkg/users/binarycaching#abi-hash

The ABI hash is determined before installing the port, and it is also used for looking up a cached artifact.

Now if the alternative port dirs have the same content, but you modify build parameters based on CURRENT_PORT_DIR, the artifacts can still share the same cache key. In the end you may not get what you expect when the artifact is restored from cache key, and you will have difficiculties to track this down because sometimes it works and it sometimes it doesn't.

So what you try to achieve shouldn't be a simple task but avoided.

**Reply by netcorefan1** [reply]
@dg0yt Many thanks for the warning. Without your comment I was going to do a big mistake. It hasn't been that long until CURRENT_PORT_DIR suddenly became undefined. I have not been able to identify the exact reason, but for sure it has to do with your warning. I had to resort to more complicated [code](https://github.com/microsoft/vcpkg/discussions/31116) for port location detection, but at least I will be free from such unexpected surprises.
It is a tutorial I decided to wrote after I saw the problems with my pull requests (#30918 and #30937) for people that still want the achieve this locally. If I was aware it was so hard, I would have aborted from the beginning, but it has become a challenge that I need to win in order to make sense of all my efforts.
